include/fx/gltf.h

```diff
@@ -1839,46 +1878,156 @@ namespace gltf
         }
     }
 
-    inline Document LoadFromBinary(std::string const & documentFilePath, ReadQuotas const & readQuotas = {})
+    inline Document LoadFromText(std::string const & documentFilePath, ReadQuotas const & readQuotas = {})
     {
         try
         {
-            std::vector<uint8_t> binary{};
+            std::ifstream file(documentFilePath);
+            if (!file.is_open())
             {
-                std::ifstream file(documentFilePath, std::ios::binary);
-                if (!file.is_open())
+                throw std::system_error(std::make_error_code(std::errc::no_such_file_or_directory));
+            }
+            return LoadFromTextStream(file, detail::GetDocumentRootPath(documentFilePath), readQuotas);
+        }
+        catch (std::system_error &)
+        {
+            throw;
+        }
+        catch (...)
+        {
+            std::throw_with_nested(invalid_gltf_document("Invalid glTF document. See nested exception for details."));
+        }
+    }
+
+    inline Document LoadFromBinaryStream(std::istream & is, std::string const & bufferRootPath = "", ReadQuotas const & readQuotas = {})
+    {
+        try
+        {
+            detail::GLBHeader header;
+            std::size_t totalSize = 0;
+            std::vector<uint8_t> binary(detail::HeaderSize);

```

Comment:
Read into a local var first so we don't have to resize the binary vector later (it's a small size but still best to avoid where possible)